docs/adding-new-platforms.md

```diff
@@ -0,0 +1,81 @@
+## Adding new platforms
+
+Depending on what type of platform you're adding, the instructions here will
+vary.
+
+### Adding a brand new platform
+
+- Create a new file for your mainboard in `platform-configs/`. For most
+  platforms this file will be called `[platform-vendor]-[platform-model].robot`.
+- Add the following at the top of your platform config:
+
+```robot
+*** Settings ***
+Resource    default.robot
+```
+
+- Copy the contents of `include/default.robot` to your platform config
+- Modify the file for your platform:
+    - Modify the settings appropriately for your mainboard
+    - Remove any unmodified lines - they will be sourced from `default.robot`
+- Add the platform configuration to `variables.robot:
+    - Create a new configuration of RTE, if you are using one, e.g.:
+
+    ```robot
+    &{RTE11}=                   ip=192.168.10.174    cpuid=02c000426621f7ea    pcb_rev=1.0.0
+    ...                         platform=apu4    board-revision=4d    env=prod
+    ...                         platform_vendor=PC Engines    firmware_type=UEFI
+    ```
+
+    - Add the RTE to the list:
+
+    ```
+    @{RTE_LIST}=                &{RTE05}
+    ...                         &{RTE06}    &{RTE07}    &{RTE08}    &{RTE09}    &{RTE10}
+    ...                         &{RTE11}
+    ```
+
+    - Do the same for any modules installed in the platform
+    - Create a new CONFIG contaning the RTE and modules used for testing, and
+      append it to the list:
+
+    ```
+    @{CONFIG04}=                &{RTE11}    &{SSD06}    &{CARD06}    &{USB03}
+    ...                         &{MODULE06}    &{ADAPTER01}    &{MODULE10}
+
+    @{CONFIG_LIST}=             @{CONFIG01}    @{CONFIG02}    @{CONFIG03}    @{CONFIG04}
+    ```
+
+    - Run a simple test to verify the config is working correctly - for example
+      UEFI Shell:
+
+    ```
+    robot -v snipeit:no -L TRACE -v rte_ip:192.168.10.174 -v device_ip:0.0.0.0 -v config:pcengines-apu4 dasharo-compatibility/uefi-shell.robot
```

Comment:
This is not an excellent canonical test to check configuration IMHO, it should be booting OS or something like that. Maybe flashing and getting into BIOS menu depending on the BIOS version

Reply:
Flashing depends on more varliables and OS booting depends on having working storage and having the storage tests configured properly. I chose UEFI shell because it seemed like the smallest effort, but I can change to something else

Reply:
What about platforms without UEFI?

Reply:
I tend to use `custom-boot-menu-key.robot` as the simplest test we can present with the OSFV.

Reply:
Changed the test to `custom-boot-menu-key.robot` in commit 67e5504a19f4a47db82f5b7f3b9f5730b0824dce